substrate/primitives/keystore/Cargo.toml

```diff
@@ -28,7 +27,7 @@ rand_chacha = "0.2.2"
 
 [features]
 default = ["std"]
-std = ["codec/std", "sp-core/std", "sp-externalities/std"]
+std = ["codec/std", "dep:parking_lot", "sp-core/std", "sp-externalities/std"]
```

Comment:
```suggestion
std = ["codec/std", "parking_lot", "sp-core/std", "sp-externalities/std"]
```

Reply:
@bkchr Hm... sure, I can do this, but just to confirm - do we want to never use the `dep:` syntax unless necessary?

I'm asking because this is not just a stylistic choice. The nice thing about the new (well, new-ish) `dep:` syntax it that it prevents an implicit feature from being defined, that is, if an `optional` dependency is *not* referred to with a `dep:` then cargo will implicitly define a new feature like this:

```
[features]
parking_lot = ["dep:parking_lot"]
```

And in consequence this now *technically* becomes part of the public API for the crate, so another crate which depends on the `sp-keystore` can then say:

```
[dependencies]
sp-keystore = { version = "...", features = ["parking_lot"] }
```

...which doesn't make much sense because the `parking_lot` dependency here is an *internal* dependency, so it makes *zero* sense for another crate to be able to enable it. And using it through `dep:` prevents this, because then `cargo` won't create this implicit feature.

Reply:
Okay good points, then let's ignore what I said. I didn't realized that this would then not be exposed :+1: